ext/drreg/drreg.c

```diff
@@ -65,17 +65,65 @@
 
 #define PRE instrlist_meta_preinsert
 
-/* This should be pretty hard to exceed as there aren't this many GPRs */
-#define MAX_SPILLS (SPILL_SLOT_MAX + 8)
+/* drreg uses DR spill slots by default if no addressable TLS
+ * storage is available. Since these slots are shared also with clients,
+ * it could be the case that the client uses a lot of these slots, resulting
+ * in an insufficient number of available slots for drreg to function properly.
+ *
+ * In order to address this concern, we use a hard-coded max on the number
+ * of spill slots tracked. This number of spills intentionally surpasses
+ * SPILL_SLOT_MAX to cater both for drreg and possible clients,
+ * It is set to be pretty hard to exceed as there aren't that many GPRs.
+ */
+#define MAX_SPILLS (SPILL_SLOT_MAX + DR_NUM_GPR_REGS)
```

Comment:
Worried that somebody might think this is an exact number. I would add a comment that it is *not* an exact number and an arbitrarily chosen upper limit of trackable slots that should accommodate all use cases.

By they way, maybe add +1 for aflags.

Reply:
Done